content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+documentation/examples, you will see -A INPUT.
+The difference is that -A appends to the list of rules,
+while -I INPUT 1 inserts before the first entry.
+Usually when adding new accept ports to iptables configuration,
+you want to put them at the beginning of the ruleset, not the end.
+Run iptables -L -n again and you should now see entries for these 4 ports.
+
+==== Forwarding
+
+Once traffic on the required ports are allowed, you can run the command
+to forward port 80 traffic to 8080, and port 443 traffic to 8443.
+The commands look like this:
+
+[source]
+----
+sudo iptables -A PREROUTING -t nat -i eth0 -p tcp --dport 80 -j REDIRECT --to-port 8080
+sudo iptables -A PREROUTING -t nat -i eth0 -p tcp --dport 443 -j REDIRECT --to-port 8443
+----
+
+You can verify the forwarding rules using below command.
+
+[source]
+----
+[root@xyz~]# iptables -L -t nat
+Chain PREROUTING (policy ACCEPT)
+target     prot opt source               destination
+REDIRECT   tcp  --  anywhere             anywhere             tcp dpt:http redir ports 8080
+REDIRECT   tcp  --  anywhere             anywhere             tcp dpt:https redir ports 8443
+
+Chain INPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain OUTPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+----
+
+ +
+
+Once these rules are set and confirmed with iptables -L -n, and once
+your Jenkins instance is up and running on port 8080, attempt to access
+your Jenkins instance on port 80 instead of 8080.
+It should work and your URL should stay on port 80 - in other words,
+it should not get redirected to 8080.
+The fact that forwarding from 80 to 8080 (or 443 to 8443) should remain
+hidden from the client.
+
+===== Saving iptables Configuration
+
+Using the iptables command to change port configuration and routing
+rules only changes the current, in-memory configuration.
+It does not persist between restarts of the iptables service.
+So, you need to make sure you save the configuration to make the changes permanent.
+
+Saving the configuration is slightly different between RedHat-based and
+Debian-based systems.
+On a RedHat-based system (Fedora, CentOS, RHEL, etc), issue the following command:
+
+[source]
+----
+sudo iptables-save > /etc/sysconfig/iptables
+----
+
+On a Debian-based system (Debian, Ubuntu, Mint, etc), issue the
+following command:
+
+[source]
+----
+sudo sh -c "iptables-save > /etc/iptables.rules"
+----
+
+The iptables-restore command will need to be executed manually, or your
+system configured to automatically run it on boot, against the
+/etc/iptables.rules file you have created, in order for your iptables
+configuration to be retained across reboots.
+On Ubuntu fastest way is to install iptables-persistent after configuring iptables -
+it will automatically create necessery files from current configuration and load
+them on boot.
```

Comment:
Phrasing changes
```suggestion
On Ubuntu, the fastest way is to install `iptables-persistent` after configuring iptables.
It will automatically create the required files from the current configuration and load them on boot.
```